src/components/ProfilePage/Leaderboard/Leaderboard.tsx

```diff
@@ -0,0 +1,82 @@
+import { Box, Stack, useTheme } from '@mui/material';
+import { NoSelectTypography } from '../ProfilePage.style';
+import { LeaderboardContainer } from './Leaderboard.style';
+import { XPIcon } from '../../../components/illustrations/XPIcon';
+import { useLeaderboardList, useLeaderboardUser } from '../../../hooks/useLeaderboard';
+import { Button } from '../../../components/Button';
+import ChevronRight from '@mui/icons-material/ChevronRight';
+import ChevronLeft from '@mui/icons-material/ChevronLeft';
+import { useEffect, useState } from 'react';
+import { LeaderboardSkeleton } from './LeaderboardSkeleton';
+
+const LEADERBOARD_LENGTH = 25;
+
+export const Leaderboard = ({ address }: { address?: string }) => {
+  const theme = useTheme();
+  
+  const [currentPage, setCurrentPage] = useState(1);
+  const { data: leaderboardData, meta } = useLeaderboardList(currentPage, LEADERBOARD_LENGTH);
+  const { data: leaderboardUserData } = useLeaderboardUser(address);
+
+  const leaderboardListLength = meta?.pagination?.pagesLength || LEADERBOARD_LENGTH;
+
+  useEffect(() => {
+    setCurrentPage(current => Math.min(Math.max(current, 1), leaderboardListLength));
+  }, [leaderboardListLength]);
+
+  const nextPage = () => {
+    setCurrentPage(current => Math.min(current + 1, leaderboardListLength));
+  };
+  
+  const previousPage = () => {
+    setCurrentPage(current => Math.max(current - 1, 1));
+  };
+
+  return (
+    <LeaderboardContainer>
+      <NoSelectTypography fontSize="14px" lineHeight="18px" fontWeight={700}>
+        RANK
+      </NoSelectTypography>
+      <NoSelectTypography
+        color={theme.palette.mode === 'light' ? theme.palette.accent1.main : theme.palette.white.main}
+        fontWeight={700}
+        sx={{ fontSize: { xs: 28, sm: 48 } }}
+      >
+        {leaderboardUserData ? leaderboardUserData.position : '-'}
```

Comment:
Can be simplify with 
```suggestion
        {leaderboardUserData?.position ?? '-'}
```